service/collector/trusted_advisor.go

```diff
@@ -0,0 +1,240 @@
+	// if Trusted Advisor is not supported (support plan is not Business or Enterprise).
+	trustedAdvisorUnsupportedErrorMessage = "AWS Premium Support Subscription is required to use this service."
+
+	regionIndex    = 0
+	serviceIndex   = 1
+	limitNameIndex = 2
+	limitIndex     = 3
+	usageIndex     = 4
+
+	regionLabel    = "region"
+	serviceLabel   = "service"
+	limitNameLabel = "name"
+)
+
+var (
+	serviceLimit *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "", "service_limit"),
+		"Service limits as reported by Trusted Advisor.",
+		[]string{
+			regionLabel,
+			serviceLabel,
+			limitNameLabel,
+		},
+		nil,
+	)
+	serviceUsage *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "", "service_usage"),
+		"Service usage as reported by Trusted Advisor.",
+		[]string{
+			regionLabel,
+			serviceLabel,
+			limitNameLabel,
+		},
+		nil,
+	)
+
+	trustedAdvisorSupport *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "", "trusted_advisor_supported"),
+		"Gauge describing whether Trusted Advisor is available with your support plan.",
+		nil, nil,
+	)
+
+	getChecksError = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: Namespace,
+		Name:      "trusted_advisor_get_checks_error_count",
+		Help:      "Counter for the number of errors encountered getting Trusted Advisor checks.",
+	})
+	getResourcesError = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: Namespace,
+		Name:      "trusted_advisor_get_resources_error_count",
+		Help:      "Counter for the number of errors encountered getting Trusted Advisor resources.",
+	})
+	convertResourcesError = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: Namespace,
+		Name:      "trusted_advisor_convert_resources_error_count",
+		Help:      "Counter for the number of errors encountered converting Trusted Advisor resources.",
+	})
+
+	getChecksDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: Namespace,
+		Name:      "trusted_advisor_get_checks_duration",
+		Help:      "Histogram for the duration of Trusted Advisor get checks calls.",
+	})
+	getResourcesDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: Namespace,
+		Name:      "trusted_advisor_get_resources_duration",
+		Help:      "Histogram for the duration of Trusted Advisor get resource calls.",
+	})
+)
+
+func init() {
+	prometheus.MustRegister(getChecksError)
+	prometheus.MustRegister(getResourcesError)
+	prometheus.MustRegister(convertResourcesError)
+
+	prometheus.MustRegister(getChecksDuration)
+	prometheus.MustRegister(getResourcesDuration)
+}
+
+func (c *Collector) collectTrustedAdvisorChecks(ch chan<- prometheus.Metric) {
+	c.logger.Log("level", "debug", "message", "collecting metrics for trusted advisor checks")
+
+	checks, err := c.getTrustedAdvisorChecks()
+	if err != nil {
+		if strings.Contains(err.Error(), trustedAdvisorUnsupportedErrorMessage) {
```

Comment:
Lets have this please with the other errors so there is just the usual error/matcher pair.

Reply:
Fixed